core/tests/Test.Microsoft.Identity.Core.Unit/Mocks/MockHttpMessageHandler.cs

```diff
@@ -54,6 +54,8 @@ internal class MockHttpMessageHandler : HttpMessageHandler
 
         public Exception ExceptionToThrow { get; set; }
 
+        public Action<HttpRequestMessage> AdditionalRequestValidation { get; set; }
```

Comment:
why do you need this ? you can validate QueryParams and PostData without this

Reply:
@trwalke thoughts?

Reply:
@trwalke

---
In reply to: [202848908](https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/pull/1124#discussion_r202848908) [](ancestors = 202848908)

Reply:
This is how ADAL is designed to check QueryParams and PostData currently. I wanted the two to be the same so that we can write tests using the same methods that we do in ADAL. How else can we verify QueryParams and PostData in unit tests?

Reply:
Followed up offline, currently, MSAL can check if the value exists in the headers but it cannot check if the parameter does not exist. The AdditionalRequestValidation will enable many more validation scenarios. (resolved)